### PR TITLE
Clean up contribution guidelines docs and add 'requirements' for D&C SIG

### PR DESCRIPTION
**Pull request by sptramer**

### CODE REVIEW

**Comment by sptramer** [comment]
Replaces #87

File: community-membership.md [hunk 1]
```diff
@@ -48,7 +60,9 @@ Requirements to continue to be a Reviewer:
 
 If a Reviewer fails to meet the requirements, they will be given a 1 month grace period.  After the 2nd month, if they haven't returned to the required number of reviews, they will be returned to Contributor status.
 
-## How to become a maintainer
+## Maintainers
+
+A *Maintainer* role is an increased level of responsibility for a Reviewer. Maintainers aren't just responsible for performing review - they also have the ability to merge pull requests into the appropriate branch. Maintainers are high-quality Reviewers who take extra care when looking at a pull request, and are willing to trust the other Reviewers who approve work.
```
Comment (NicholasVanSickle) [inline community-membership.md:65]: Should we maybe more explicitly call out that it's not just the ability to merge PRs, but also the corresponding responsibility for managing PRs as they merge in?

File: community-membership.md [hunk 2]
```diff
@@ -6,39 +6,81 @@ The [O3DE community repo](https://github.com/o3de/community) contains informatio
...
 To contribute, please review our [Code of Conduct](https://github.com/o3de/o3de/blob/development/CODE_OF_CONDUCT.md) first. 
 
-## How to become a reviewer
+## Contributing to Open 3D Engine
 
-Requirements to become a reviewer for a given Special Interest Group
-* 6+ contributions successfully submitted to O3DE
+Everyone is invited and welcome to contribute to Open 3D Engine by filing issues, pull requests, asking questions, participating in discussions and feature reviews, and engaging with _any_ public work. All you need to do is agree to the [Code of Conduct](https://github.com/o3de/o3de/blob/development/CODE_OF_CONDUCT.md), ensure that your contribution is valid under the [Developer Certificate of Origin](https://www.dita-ot.org/dco) and sign your commits. Discussion requires no such agreement.
+
+For contributors who are interested in working more in-depth on features or writing specifications, providing specific content types like videos or blogs, or managing groups of work, contact the appropriate [O3DE Special Interest Group]() to find out about their process and how to meet any requirements for official contributions.
```
Comment (lmbr-pip) [inline community-membership.md:13]: nit: seems over wordy. This is really to just to highlight how to contribute in specific areas, I would just call out that we have SIGs including technical, marketting and docs and link to https://github.com/o3de/community/blob/main/CONTRIBUTING.md which covers SIG areas of ownership

File: community-membership.md [hunk 3]
```diff
@@ -6,39 +6,81 @@ The [O3DE community repo](https://github.com/o3de/community) contains informatio
...
+
+* *Code Reviewers*, responsible for reviewing source code submitted to repositories maintained by the Open 3D Foundation.
+* *Documentation Reviewers*, responsible for reviewing items under the [Documentation & Community Special Interest Group charter](https://github.com/o3de/sig-docs-community/blob/main/governance/charter.md).
+* *Technical Reviewers*, code reviewers who also review documentation for technical accuracy.
+
+You must be nominated (self-nomination is allowed) for promotion from Contributor to Reviewer. To do this, an issue should be created on the SIG's repo for the supportive Reviewers and Maintainers to comment and add their support. For example, to nominate a reviewer on behalf of sig-content, an Issue should be created in the [sig-content repository](https://github.com/o3de/sig-content/issues/new/choose).
+
+Once you have the requisite support, the nomination Issue needs to be reviewed and approved during a SIG meeting. If possible, the nominee should attend and add it to the agenda for the SIG to review and, if appropriate, approve. Nomination Issues that do not have the necessary support will be closed after 4 weeks of inactivity.
```
Comment (lmbr-pip) [inline community-membership.md:25]: nit: captilization, either "... Nomination Issue .." or "nomination issue"

"during a SIG meeting. The nominee should add their nomination issue to the agenda to an upcoming meeting for the relevant and, where possible attend that meeting, to get SIG approval for their nomination.

File: community-membership.md [hunk 4]
```diff
@@ -6,39 +6,81 @@ The [O3DE community repo](https://github.com/o3de/community) contains informatio
...
+The specialized D&C SIG role of _Documentation Technical Reviewer_ requires:
+
+* Being a Reviewer or Maintainer for a SIG governing the appropriate technology.
+* 2+ of the appropriate SIG's Reviewers or Maintainers who support the promotion.
+
+### Keeping Reviewer status
+
+Requirements to continue to be a Reviewer:
+
+* 4+ Pull Requests reviewed per month
```
Comment (lmbr-pip) [inline community-membership.md:59]: Is this possible for SIGs with low volume? Esp where there are multiple document technical reviews per SIG.

File: community-membership.md [hunk 5]
```diff
@@ -6,39 +6,81 @@ The [O3DE community repo](https://github.com/o3de/community) contains informatio
...
+
+If a Reviewer fails to meet the requirements, they will be given a 1 month grace period.  After the 2nd month, if they haven't returned to the required number of reviews, they will be returned to Contributor status.
+
+## Maintainers
+
+A *Maintainer* role is an increased level of responsibility for a Reviewer. Maintainers aren't just responsible for performing review - they also have the ability to merge pull requests into the appropriate branch. Maintainers are high-quality Reviewers who take extra care when looking at a pull request, and are willing to trust the other Reviewers who approve work.
+
+You must be nominated (self-nomination is allowed) for promotion from Reviewer to Maintainer. To do this, an issue should be created on the community repo for the supportive Maintainers to comment and add their support. Once you have the requisite support, the nomination Pull Request needs to be reviewed and approved during a SIG meeting.  The nominee should plan to attend and add it to the agenda for the SIG to review and, if appropriate, approve. Nomination Pull Requests that do not have the necessary support will be closed after 4 weeks of inactivity.
```
Comment (lmbr-pip) [inline community-membership.md:67]: Once you have the requisite support, the nomination issue needs to be reviewed and approved during a SIG meeting. The nominee should add the issue to an upcoming agenda for the relevant SIG and ideally attend that meeting. The SIG will then review and if appropriate, approve the status change.

**Comment by lmbr-pip** [comment]
@sptramer Do we have a plan to deliver this PR or does it need to be closed?

**Pull request closed without merging**